src/main/java/org/reaktivity/nukleus/http2/internal/types/stream/GoawayFW.java

```diff
@@ -107,9 +88,22 @@ public int limit()
     }
 
     @Override
-    public Http2GoawayFW wrap(DirectBuffer buffer, int offset, int maxLimit)
+    public GoawayFW wrap(DirectBuffer buffer, int offset, int maxLimit)
     {
         super.wrap(buffer, offset, maxLimit);
+
+        int streamId = Http2FrameFW.streamId(buffer, offset);
+        if (streamId != 0)
+        {
+            throw new Http2Exception(String.format("Invalid stream-id=%d for GOAWAY frame", streamId));
```

Comment:
We don't need a custom unchecked exception type, something like `IllegalArgumentException` is sufficient here.